actor/v7action/service_instance_test.go

```diff
@@ -368,6 +363,239 @@ var _ = Describe("Service Instance Actions", func() {
 		})
 	})
 
+	Describe("UpdateManagedServiceInstance", func() {
+		const (
+			serviceInstanceName = "fake-service-instance-name"
+			guid                = "fake-service-instance-guid"
+			spaceGUID           = "fake-space-guid"
+		)
+
+		When("the service instance is successfully updated synchronously", func() {
+			BeforeEach(func() {
+				fakeCloudControllerClient.GetServiceInstanceByNameAndSpaceReturns(
+					resources.ServiceInstance{
+						Type: resources.ManagedServiceInstance,
+						GUID: guid,
+					},
+					ccv3.IncludedResources{},
+					ccv3.Warnings{"warning from get"},
+					nil,
+				)
+				fakeCloudControllerClient.UpdateServiceInstanceReturns(
+					"",
+					ccv3.Warnings{"warning from update"},
+					nil,
+				)
+			})
+
+			It("makes the right calls and returns all warnings", func() {
+				warnings, err := actor.UpdateManagedServiceInstance(
+					serviceInstanceName,
+					spaceGUID,
+					resources.ServiceInstance{Tags: types.NewOptionalStringSlice("foo", "bar")},
+				)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(warnings).To(ConsistOf("warning from get", "warning from update"))
+
+				Expect(fakeCloudControllerClient.GetServiceInstanceByNameAndSpaceCallCount()).To(Equal(1))
```

Comment:
I'm a big fan of `By` to divide tests with multiple related but different assertions like this one. I know it's a matter of preference, but it's something to consider 👀

Reply:
I’ll look into doing that for the next story